### PR TITLE
Add flag to configure the prometheus job name to be used when fetching the history data

### PR DESCRIPTION
**Pull request by mrIncompetent**

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/recommender/input/history/history_provider_test.go [hunk 1]
```diff
@@ -48,7 +48,8 @@ func (m *mockPrometheusClient) GetTimeseries(query string) ([]Timeseries, error)
 func TestGetEmptyClusterHistory(t *testing.T) {
 	mockClient := mockPrometheusClient{}
 	historyProvider := prometheusHistoryProvider{
-		prometheusClient: &mockClient}
+		prometheusCadvisorJobName: "kubernetes-cadvisor",
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/recommender/input/history/history_provider_test.go]: Can you make the job name a const?

Comment (mrIncompetent) [reply vertical-pod-autoscaler/pkg/recommender/input/history/history_provider_test.go]: done

File: vertical-pod-autoscaler/pkg/recommender/main.go [hunk 2]
```diff
@@ -31,13 +31,14 @@ import (
...
-	prometheusAddress      = flag.String("prometheus-address", "", `Where to reach for Prometheus metrics`)
-	storage                = flag.String("storage", "", `Specifies storage mode. Supported values: prometheus, checkpoint (default)`)
-	address                = flag.String("address", ":8942", "The address to expose Prometheus metrics.")
-	kubeApiQps             = flag.Float64("kube-api-qps", 5.0, `QPS limit when making requests to Kubernetes apiserver`)
-	kubeApiBurst           = flag.Float64("kube-api-burst", 10.0, `QPS burst limit when making requests to Kubernetes apiserver`)
+	metricsFetcherInterval    = flag.Duration("recommender-interval", 1*time.Minute, `How often metrics should be fetched`)
+	checkpointsGCInterval     = flag.Duration("checkpoints-gc-interval", 10*time.Minute, `How often orphaned checkpoints should be garbage collected`)
+	prometheusAddress         = flag.String("prometheus-address", "", `Where to reach for Prometheus metrics`)
+	prometheusCAdvisorJobName = flag.String("prometheus-cadvisor-job-name", "kubernetes-cadvisor", `Name of the prometheus job name which scrapes the cAdvisor metrics`)
```
[the remark this answers is not in the document]
Comment (mrIncompetent) [reply vertical-pod-autoscaler/pkg/recommender/main.go]: I renamed it to `prometheusJobName`

**Comment by bskiba** [comment]
Thanks!
/lgtm
/approve

**Pull request merged**